yb-voyager/cmd/assessMigrationCommandBulk.go

```diff
@@ -185,41 +204,73 @@ func parseFleetConfigFile(filePath string) ([]AssessMigrationDBConfig, error) {
 	}
 	defer file.Close()
 
-	scanner := bufio.NewScanner(file)
-	for scanner.Scan() {
-		line := scanner.Text()
-		if len(line) == 0 {
-			continue
-		}
-		dbConfig := parseFleetConfigLine(line)
-		dbConfigs = append(dbConfigs, dbConfig)
+	reader := csv.NewReader(file)
+	header, err := reader.Read()
+	if err != nil {
+		return nil, fmt.Errorf("failed to read fleet config file header: %w", err)
 	}
+	header = normalizeFleetConfFileHeader(header)
+
+	lineNum := 2
+	for {
+		record, err := reader.Read()
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			return nil, fmt.Errorf("failed to read line %d: %w", lineNum, err)
+		}
 
-	if err := scanner.Err(); err != nil {
-		return nil, err
+		if len(record) != len(header) {
+			return nil, fmt.Errorf("field count mismatch on line %d: expected %d fields but got %d", lineNum, len(header), len(record))
+		}
+
+		dbConfig, err := createDBConfigFromRecord(record, header)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create config for line %d in fleet config file: %w", lineNum, err)
+		}
+		dbConfigs = append(dbConfigs, *dbConfig)
+		lineNum++
 	}
 
 	return dbConfigs, nil
 }
 
-/*
-Format: covers both SSL and non-SSL cases
+func createDBConfigFromRecord(record []string, header []string) (*AssessMigrationDBConfig, error) {
+	configMap := make(map[string]string)
+	for i, field := range header {
+		configMap[field] = strings.TrimSpace(record[i])
+	}
 
-	<dbtype>,<hostname>,<port>,<service_name>,<sid>,<tns_alias>,<username>,<password>,<schema>
-*/
-func parseFleetConfigLine(line string) AssessMigrationDBConfig {
-	config := strings.Split(line, ",")
-	return AssessMigrationDBConfig{
-		DbType:      config[0],
-		Host:        config[1],
-		Port:        config[2],
-		ServiceName: config[3],
-		SID:         config[4],
-		TnsAlias:    config[5],
-		User:        config[6],
-		Password:    config[7],
-		Schema:      config[8],
+	// Check that at least one of the mandatory field combinations is non-empty and valid
+	validConfig := false
+	for _, mandatoryCombo := range mandatoryHeaderFields {
+		missingOrEmptyFields := false
+		for _, field := range mandatoryCombo {
+			if val, ok := configMap[field]; !ok || val == "" {
+				missingOrEmptyFields = true
+				break
+			}
+		}
+		if !missingOrEmptyFields {
+			validConfig = true
+			break
+		}
 	}
+	if !validConfig {
+		return nil, fmt.Errorf("record does not contain a valid combination of mandatory fields")
```

Comment:
let's give more information here. Which mandatory fields are missing.. and in which row.

Reply:
done